lib/compeon/token/base.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+module Compeon
+  module Token
+    module Base
+      def self.attributes(**attributes_mapping)
```

Comment:
I love this structure for our services in marketplace, but I'm not sure about introducing it here, in a relatively small gem. Why did you go that route? Couldn't a similar behavior (talking about the methods in `class_eval`) be achieved with inheritence?

Reply:
Timo and I discussed this early this morning. IMO this approach is way too complex for the amount of token classes we expect to have (and how much they will differ from each other). I would advocate for a mix of inheritance and composition (concerns) for this use case.